util/oidc/oidc.go

```diff
@@ -506,3 +568,38 @@ func createClaimsAuthenticationRequestParameter(requestedClaims map[string]*oidc
 	}
 	return oauth2.SetAuthURLParam("claims", string(claimsRequestRAW)), nil
 }
+
+// GetUserInfo queries the IDP userinfo endpoint for claim
+func (a *ClientApp) GetUserInfo(accessToken, issuer, userInfoPath string) (claims Claims, unauthorized bool, err error) {
+	url := issuer + userInfoPath
+	request, err := http.NewRequest("GET", url, nil)
+
+	if err != nil {
+		err = fmt.Errorf("failed creating new http request: %w", err)
+		return
+	}
+
+	bearer := fmt.Sprintf("Bearer %s", accessToken)
+	request.Header.Set("Authorization", bearer)
+
+	response, err := httpClient.Do(request)
+
+	if response.StatusCode == http.StatusUnauthorized {
+		unauthorized = true
+		return
+	}
+	if err != nil {
+		err = fmt.Errorf("failed to query userinfo endpoint of IDP: %w", err)
+		return
+	}
+
+	defer response.Body.Close()
+	err = json.NewDecoder(response.Body).Decode(&claims)
```

Comment:
yeah I have seen the comment from @zachaller. Glancing at argo-workflows, I don´t see them having implemented it. I could try to do it, but as I have no experience with it, it might take a little bit longer.

So I'd support the idea of skipping it for now and either mark as follow up or mention somewhere in the code...

Reply:
Gotcha. Will signed responses always fail JSON deserialization? If not, I think we should explicitly reject signed responses, so we remain compliant with the spec.

Reply:
I think it will fail deserialization as the http body doesn't contain valid JSON directly. I can see if there is a flag to reject signed responses, but yeah not later than then it will certainly fail if the response was signed..

Reply:
Okay so I found out that then the Content-type that is returned as response is `application/jwt`, it's actually signed. So I just copied the code from the first validation down and now the UserInfo response seems validated.